modules/git/git_test.go

```diff
@@ -1,14 +1,89 @@
 package git
 
 import (
+	"fmt"
+	"os"
+	"os/exec"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestGetCurrentBranchName(t *testing.T) {
-	t.Parallel()
+func TestGetCurrentBranchNameReturnsBranchName(t *testing.T) {
+	err := exec.Command("git", "checkout", "master").Run()
+	assert.Nil(t, err)
```

Comment:
Can you actually clone the repo to a new temp dir and then run all of these tests in a subtest? That way, it doesn't have the surprising effect of changing what branch you are working on which can cause problems in CI when running all the tests.

Something like:

```go
import (
    "testing"
    "io/ioutil"
    "path/filepath"

    "github.com/gruntwork-io/terratest/modules/shell"
)

func TestGitRefChecks(t *testing.T) {
    t.Parallel()

    tmpdir, err := ioutil.TempDir("", "")
    require.NoError(t, err)
    defer os.RemoveAll(tmpdir)
    gitWorkDir := filepath.Join(tmpdir, "terratest")

    cmd := shell.Command{
        Command: "git",
        Args: []string{"clone", "https://github.com/gruntwork-io/terratest.git", gitWorkDir},
    }
    shell.RunCommand(t, cmd)

    t.Run("GetCurrentBranchNameReturnsBranchName", testGetCurrentBranchNameReturnsBranchName)
    t.Run("GetCurrentBranchNameReturnsEmptyForDetachedState", testetCurrentBranchNameReturnsEmptyForDetachedState)
    // ... and so on ...
}
```

Reply:
I think I've done exactly that (sorry if i misunderstood).
Temp repo is cloned within the setup function which is run before whole test module (cleanup is added also).

Please check:
https://github.com/gruntwork-io/terratest/pull/588/files#diff-1b9f8aa5feac10022b44c29ce84f78e1R57-R70

Reply:
Ahhhh sorry I missed that when quickly glancing.

With that said, we prefer the pattern of using test functions and subtests as opposed to test main. Can you switch the pattern so you are using subtests instead as shown above?

Reply:
Sure thing. Switched to subtests, ready for another review :)